box_error_msgpack_test.go

```diff
@@ -0,0 +1,314 @@
+package tarantool_test
```

Comment:
In general, it could be better tested, although it is meditative.

https://coveralls.io/builds/54505878/source?filename=box_error.go

Reply:
For now there are no similar tests in common-purpose decoding:

https://coveralls.io/builds/54545306/source?filename=response.go

But I can try to make it if you wish.

Reply:
Added tests for possible msgpack decode corner cases. Has zero idea how to cover possible encode errors. (I think the only way to fail this is to mess with buffer size, which is purely internal.)